pkg/target/host.go

```diff
@@ -182,8 +204,26 @@ func (h *Host) Connect() {
 	}
 
 	h.Conn = conn
+	h.updateHostHealthStatus <- &HostStatus{Host: h, Status: true, sigCh: make(chan struct{})}
 
 	h.Wm.Lock()
 	defer h.Wm.Unlock()
 	h.W = bufio.NewWriterSize(conn, h.bufSize)
 }
+
+func (h *Host) getConnectionToHost() (net.Conn, error) {
+	dialer := net.Dialer{
+		Timeout:   time.Duration(h.ConnTimeoutSec) * time.Second,
+		KeepAlive: time.Duration(h.KeepAliveSec) * time.Second,
+	}
+	conn, err := dialer.Dial("tcp", fmt.Sprintf("%s:%d", h.Name, h.Port))
+	return conn, err
+}
+
+func (h *Host) checkUpdateHostStatus() {
+	conn, _ := h.getConnectionToHost()
```

Comment:
This will probably need a bit of jitter. 
Not saying we have do it now, let's get something working and then we can improve it.